tangelo/toolboxes/operators/operators.py

```diff
@@ -20,17 +20,102 @@
+                raise RuntimeError("Spin must be the same for all FermionOperators.")
+
+        if isinstance(other, FermionOperator):
+            return super(FermionOperator, self).__imul__(other)
+        elif isinstance(other, of.FermionOperator):
+            if self.n_spinorbitals is not None or self.n_electrons is not None or self.spin is not None:
+                raise RuntimeError("openfermion FermionOperator did not define a necessary attribute")
+            else:
+                f_op = FermionOperator()
+                f_op.terms = other.terms.copy()
+                return super(FermionOperator, self).__imul__(f_op)
+        else:
+            return super(FermionOperator, self).__imul__(other)
+
+    def __mul__(self, other):
+        return self.__imul__(other)
+
+    def __rmul__(self, other):
+        return self.__imul__(other)
+
+    def __iadd__(self, other):
+        # Raise error if attributes are not the same across Operators.
+        if (self.n_spinorbitals is not None and other.n_spinorbitals is not None)\
+                or (self.n_electrons is not None and other.n_electrons is not None)\
+                or (self.spin is not None and other.spin is not None):
+
+            if self.n_spinorbitals != other.n_spinorbitals:
+                raise RuntimeError("n_spinorbitals must be the same for all FermionOperators.")
+            elif self.n_electrons != other.n_electrons:
+                raise RuntimeError("n_electrons must be the same for all FermionOperators.")
+            elif self.spin != other.spin:
+                raise RuntimeError("Spin must be the same for all FermionOperators.")
+
+        if isinstance(other, FermionOperator):
+            return super(FermionOperator, self).__iadd__(other)
+        elif isinstance(other, of.FermionOperator):
+            if self.n_spinorbitals is not None or self.n_electrons is not None or self.spin is not None:
+                raise RuntimeError("openfermion FermionOperator did not define a necessary attribute")
+            else:
+                f_op = FermionOperator()
+                f_op.terms = other.terms.copy()
+                return super(FermionOperator, self).__iadd__(f_op)
+
+    def __add__(self, other):
+        return self.__iadd__(other)
+
+    def __radd__(self, other):
+        return self.__iadd__(other)
+
+    def __isub__(self, other):
+        return self.__iadd__(-1. * other)
+
+    def __sub__(self, other):
+        return self.__isub__(other)
+
+    def __rsub__(self, other):
+        return self.__isub__(other)
+
+    def __eq__(self, other):
+        # Additional checks for == operator.
+        if isinstance(other, of.FermionOperator):
+            return super(FermionOperator, self).__eq__(other)
+        elif (self.n_spinorbitals != other.n_spinorbitals) or (self.n_electrons != other.n_electrons) or (self.spin != other.spin):
+            return False
+
+        return super(FermionOperator, self).__eq__(other)
```

Comment:
This statement does not make sense. Here's why:

For this statement to work, `self` must be a child class of `FermionOperator` (otherwise super does not work.)
But if that was the case, we fall into the branch at L112 and we return at L113. Therefore never hitting L117.